### PR TITLE
DDR5 support for pdata

### PR DESCRIPTION
**Pull request by aravynd**
Changes to the MRW parser to handle DDR5 DIMMs

### CODE REVIEW

File: scripts/genDTS.pl [hunk 1]
```diff
@@ -1,4 +1,4 @@
-#!/usr/bin/env perl
```
Comment (RameshIyyar) [inline scripts/genDTS.pl]: I think its accidentally changed, please fix it, and also please check other scripts that pushed in this PR

File: scripts/processMrw_bmc.pm [hunk 2]
```diff
@@ -48,8 +50,34 @@ sub buildBMCAffinity
...
+                ($target_type eq "unit-ddr5-jedec"))
+               {
+                   next;
+               }
+
             my $location_code = getLocationCode($targetObj, $target);
+            # needed for Bonnell which has older DIMM type
             $targetObj->setAttribute($target, "LOCATION_CODE", $location_code);
+
+            foreach my $child (@{ $targetObj->getTargetChildren($target) })
```
Comment (RameshIyyar) [inline scripts/processMrw_bmc.pm:71]: It's unclear to me, why are adding a LOCATION_CODE attribute to the unit targets. what is the use case?

Please point to the generated device tree for this.

File: scripts/BusFruCallouts.pm [hunk 3]
```diff
@@ -238,29 +238,24 @@ sub setupBusWithParentTarget
 
                     if ($parent eq '')
                     {
-                        # Look for the exception of an OCMB on the planar
-                        my $die = "TRUE";
-                        my $conn_target = $conn->{$busPath};
+                        # Look for the exception case of a valid OCMB target being on the planar.
```
Comment (RameshIyyar) [inline scripts/BusFruCallouts.pm:241]: I hope we are following [this](https://github.com/open-power/pdata/pull/27#discussion_r1209675362) comment to check the hw-isolation impacts.

**Comment by RameshIyyar** [comment]
@aravynd Can you please point local path to refer to the generated device tree with this PR change for reference?

File: scripts/processMrw_bmc.pm (hunk 2 above)
Comment (aravynd) [reply scripts/processMrw_bmc.pm:71]: Refer /home/aravnair/sb_san5/repo/code/open-power/1060/forsdkbuild/patch
 You will see txt files which are attribute bumps with and without the changes plus the dts files.

**Review by dcrowell77 — APPROVED** [review]

File: scripts/BusFruCallouts.pm (hunk 3 above)
Comment (aravynd) [reply scripts/BusFruCallouts.pm:241]: We have a story to handle the hw isolation changes.

File: scripts/processMrw_bmc.pm (hunk 2 above)
Comment (RameshIyyar) [reply scripts/processMrw_bmc.pm:71]: I have generated the device tree with this PR and noticed that the LOCATION_CODE attribute was added to the `dimm` target only

I thought we were adding one more child target as `unit_ddr` and in that adding the LOCATION_CODE attribute but, unit_ddr was treated as a dimm target so my comment is not valid.

File: data/p10/pdbg_compatible_propMapping.lsv [hunk 4]
```diff
@@ -24,6 +24,9 @@ unit-mem_port:ibm,power10-memport
 lcard-dimm-ddimm:ibm,power10-dimm
 lcard-dimm-ddimm4u:ibm,power10-dimm
 lcard-dimm-ddr4:ibm,power10-dimm
+unit-ddr4-jedec:ibm,power10-dimm
+unit-ddr5-jedec:ibm,power10-dimm
+unit-ddr-jedec:ibm,power10-dimm
```
Comment (RameshIyyar) [inline data/p10/pdbg_compatible_propMapping.lsv]: I think it should be `unit-ddr` instead of `unit-ddr-jedec`

**Review by RameshIyyar — CHANGES_REQUESTED** [review]
One specific thing I noticed while generating the device tree with this PR.

In the rainier 4u device tree, the `power_ic` target is added since its new target type is `chip-vreg-generic` which is treated as PMIC so the question is, whether we need to call it `power_ic`? If `power_ic` then we need to fix the compatible = "ibm,power10-pmic"; for the `power_ic` target

and also, is it guardable? if yes, we need some support from the hw-isolation as posted in the old PR comment.

I think it can tracked separately since the BMC is not using the `power_ic` target.

File: scripts/genDTS.pl (hunk 1 above)
Comment (aravynd) [reply scripts/genDTS.pl]: Done

File: data/p10/pdbg_compatible_propMapping.lsv (hunk 4 above)
Comment (aravynd) [reply data/p10/pdbg_compatible_propMapping.lsv]: Done

**Review by RameshIyyar — APPROVED** [review]

**Comment by dcrowell77** [comment]
> In the rainier 4u device tree, the `power_ic` target is added since its new target type is `chip-vreg-generic` which is treated as PMIC so the question is, whether we need to call it `power_ic`? 

Yes, it should be called power_ic.

> and also, is it guardable? if yes, we need some support from the hw-isolation as posted in the old PR comment.

Yes, it is guardable.

File: scripts/processMrw_bmc.pm (hunk 2 above)
Comment (dcrowell77) [reply scripts/processMrw_bmc.pm:71]: > I thought we were adding one more child target as `unit_ddr` and in that adding the LOCATION_CODE attribute but, unit_ddr was treated as a dimm target so my comment is not valid.

The "unit_ddr" from the MRW *is* the DIMM target on the platform.

**Pull request merged**